### PR TITLE
Update fmtlib

### PR DESCRIPTION
**Pull request by achamayou**

### CODE REVIEW

File: src/node/channels.h [hunk 1]
```diff
@@ -102,7 +102,7 @@ namespace ccf
         LOG_FAIL_FMT(
           "Invalid nonce, possible replay attack, received:{}, last_seen:{}, "
           "recv_nonce.tid:{}",
-          recv_nonce.nonce,
+          reinterpret_cast<uint64_t>(recv_nonce.nonce),
```
Comment (eddyashton) [inline src/node/channels.h:105]: Why is this needed?

Comment (achamayou) [reply src/node/channels.h:105]: Because taking non-const refs to bitfields is illegal. I'm guessing that fmtlib's type resolution has changed in a way that's now stricter, and this was presumably previously coerced somehow.

Comment (eddyashton) [reply src/node/channels.h:105]: Ah cool - I'd missed that this was a bitfield.
